samples/sample_multi_transcode/src/sample_multi_transcode.cpp

```diff
@@ -358,90 +359,98 @@ void Launcher::DoTranscoding()
 {
     mfxStatus sts = MFX_ERR_NONE;
 
-    // get parallel sessions parameters
-    mfxU32 totalSessions = (mfxU32)m_pSessionArray.size();
-    MSDKThread* pthread = 0;
-
-    for (mfxU32 i = 0; i < totalSessions; i++)
+    // Run transcoding threads
+    for (auto context : m_pThreadContextArray)
     {
-        pthread = new MSDKThread(sts, TranscodeRoutine, (void *)m_pSessionArray[i]);
-        m_HDLArray.push_back(pthread);
-    }
-
-    // Need to determine overlay threads count first
-    size_t nOverlayThreads = 0;
-    for (size_t i = 0; i < m_pSessionArray.size(); i++)
-    {
-        if (m_pSessionArray[i]->pPipeline->IsOverlayUsed())
-            nOverlayThreads++;
+        MSDK_CHECK_POINTER_NO_RET(context);
+        context->RunTranscodeRoutine();
     }
 
     // Transcoding threads waiting cycle
-    size_t aliveSessionsCount = totalSessions;
+    size_t aliveSessionsCount = m_pThreadContextArray.size();
     while (aliveSessionsCount)
     {
         aliveSessionsCount = 0;
-        for (mfxU32 i = 0; i < totalSessions; i++)
+        size_t aliveOverlaySessionsCount = 0;
+
+        for (size_t i = 0; i < m_pThreadContextArray.size(); i++)
         {
-            if (m_HDLArray[i])
-            {
-                aliveSessionsCount++;
+            if (!m_pThreadContextArray[i]->handle.valid())
+                continue;
 
-                sts = m_HDLArray[i]->TimedWait(1);
-                if (sts <= 0)
+            auto waitSts = m_pThreadContextArray[i]->handle.wait_for(std::chrono::milliseconds(1));
```

Comment:
I am starting to feel that this should be an Wait() API of ThreadTranscodeContext().

Reply:
It would be tiny wrapper, it should accept amount in ms, so in fact it will decrease
```auto waitSts = m_pThreadContextArray[i]->handle.wait_for(std::chrono::milliseconds(1));```
to
```auto waitSts = m_pThreadContextArray[i]->Wait(std::chrono::milliseconds(1));```

So it's nice to have but hardly worth it because there should be additional checks inside to be sure that future is not in detach state and so on